src/api/exchange/ExchangeApi.ts

```diff
@@ -101,21 +115,175 @@ export interface Order {
   remainingAmount: BN
 }
 
+/**
+ * BaseTradeEvent uses only info available on the emitted Trade event
+ */
+export interface BaseTradeEvent {
```

Comment:
Why do we need to redeclare the type?

In dex-telegram we have the typed event
https://github.com/gnosis/dex-telegram/blob/develop/src/services/DfusionService.ts#L166

Reply:
2 main reasons:
1. I want to decouple the smart contract event type from the API return type
2. Initially I thought I could add more info like batchId, time, and so on, but that is not available here. In the end, the only custom filed I added is `id`, which is simply `txHash | txIndex`

Wdyt, should it stay or should it go?

Reply:
Oh, I'm also doing the amount conversions from string to BN.